homeassistant/components/dsmr/config_flow.py

```diff
@@ -290,9 +308,14 @@ async def async_validate_dsmr(
         data = input_data
 
         try:
-            info = await _validate_dsmr_connection(self.hass, data)
+            is_rfxtrx = False
+            try:
+                info = await _validate_dsmr_connection(self.hass, data, is_rfxtrx)
+            except CannotCommunicate:
+                is_rfxtrx = True
+                info = await _validate_dsmr_connection(self.hass, data, is_rfxtrx)
```

Comment:
This should again be wrappen in a try except block. Could as well be that config is wrong and both option won't work.